### PR TITLE
modified fidelity

### LINKED ISSUE #179: compute sample-per-sample fidelity and gradients in recovery

### PR DESCRIPTION
**Pull request by romanjacome99**
Modified fidelity to be computed sample per sample

### CODE REVIEW

File: colibri/recovery/terms/fidelity.py [hunk 1]
```diff
@@ -44,8 +47,9 @@ def grad(self, x, y, H=None, transform=None):
             torch.Tensor: Gradient of the L1 fidelity term. 
         '''
         x = x.requires_grad_()
-        return torch.autograd.grad(self.forward(x,y, H), x, create_graph=True)[0]
-
+        norm = self.forward(x,y,H)
+ 
+        return torch.autograd.grad(norm, x, create_graph=True, grad_outputs=torch.ones_like(norm))[0]
```
Comment (bemc22) [inline colibri/recovery/terms/fidelity.py:52]: question, this formula ensures a way of "partial differentaiton" along batch?

**Review by bemc22 — CHANGES_REQUESTED** [review]
just a small clarification about autograd, LGTM

File: colibri/recovery/terms/fidelity.py (hunk 1 above)
Comment (romanjacome99) [reply colibri/recovery/terms/fidelity.py:52]: Yes, it avoids computing a Jacobian as the norm returns a vector,
